src/APU.cpp

```diff
@@ -270,6 +252,7 @@ void APU::updateChannel1() {
 
 		if (channel1.lengthCounter == 0) {
 			channel1.soundOn = 0;
+			return;
```

Comment:
Since I use soundOn in the final mix, it's fine to early out here. When the sound is restarted the channel state is reset so it doesn't matter what the state is when the channel is disabled